### PR TITLE
collection list,add,update,delete

### PR DESCRIPTION
**Pull request by hasanhameed07**
collection list,add,update,delete

### CODE REVIEW

File: crust/controllers/collection.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        data: response,
+        meta: [{length: response.length}]
+      });
+    }).catch(next);
+}
+
+function createCollection(req, res, next) {
+  collectionService.createCollection(req.body, req.params.appId)
+    .then((response) => {
+      res.status(201).json(response[0]);
```
Comment (faizulhaque-tp) [inline crust/controllers/collection.js]: any reason for response[0] ? should be handle in createCollection

File: lib/facade/collection.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  };
+  return validator.validatePayload([
+      {
+        value: payload.name,
+        rules: validator.rules.appName
+      }
+      // No validation for rules as per requirement
+    ])
+    .then(collectionRepo.create.bind(this, payload, appId, reqData.userId))
+    .then(function(response){
```
Comment (faizulhaque-tp) [inline lib/facade/collection.js]: this is extra, no need to add another then.

File: lib/facade/collection.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    _id: collectionId,
+    name: reqData.name,
+    rules: {
+      create: reqData.rules.create,
+      update: reqData.rules.update,
+      delete: reqData.rules.delete
+    },
+    updatedAt: new Date().getTime()
+  };
+  return validator.validatePayload([
```
Comment (faizulhaque-tp) [inline lib/facade/collection.js]: are you planning to add duplicate name validation within app domain?

Comment (hasanhameed07) [inline lib/facade/collection.js]: yes.. will add the validation

File: lib/facade/collection.js (hunk 2 above)
Comment (hasanhameed07) [inline lib/facade/collection.js]: yeah!

File: crust/controllers/collection.js (hunk 1 above)
Comment (hasanhameed07) [inline crust/controllers/collection.js]: okay, will move it to facade

**Comment by hasanhameedtp** [comment]
@faizulhaque-tp  Code is updated, please review and merge.

File: crust/middleware/app.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+module.exports = {
+  verifyApp
+};
+
+function verifyApp(req, res, next) {
+  req.checkParams('appId', 'Invalid AppId').isNumeric().len(15,15);
+  var errors = req.validationErrors();
+  if (errors) {
+    next(new errorHandler.badRequest(errors));
```
Comment (faizulhaque-tp) [inline crust/middleware/app.js]: "return" should be on same line

File: crust/middleware/app.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    return;
+  }
+  let appId = req.params.appId;
+  App.findOne({_id: appId, userId: req.body.userId}, '_id', function (err, result) {
+    if (result===null) {
```
Comment (faizulhaque-tp) [inline crust/middleware/app.js:18]: if(results) will do the trick :)

File: crust/routes/collection.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+const express = require('express');
+const router = express.Router();
+const collection = require('../controllers/collection');
+const authenticateMiddleware = require('../middleware/auth');
+const appMiddleware = require('../middleware/app');
+
+router.use(authenticateMiddleware.verifyToken);
+router.get('/app/:appId/collection', appMiddleware.verifyApp, collection.getCollectionListing);
+router.post('/app/:appId/collection', appMiddleware.verifyApp, collection.createCollection);
+router.post('/app/:appId/collection/:collectionId/delete', appMiddleware.verifyApp, collection.deleteCollection);
```
Comment (faizulhaque-tp) [inline crust/routes/collection.js:11]: method should be .delete
and "/delete" should be removed

File: lib/facade/collection.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  return validator.validatePayload([{
+        value: payload.name,
+        rules: validator.rules.appName
+      }
+      // No validation for rules as per requirement
+    ])
+    .then(collectionRepo.create.bind(this, payload, appId, reqData.userId))
+    .then(function(response) {
+      return response[0];
+    }).catch(function(error) {
```
Comment (faizulhaque-tp) [inline lib/facade/collection.js]: controllers already have .catch, no need to attach here.

File: lib/repositories/collection.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+          _id: appId,
+          userId: userId
+        }, {
+          $push: {
+            collections: payload._id
+          }
+        })
+      ])
+    })
+    .catch(function(error) {
```
Comment (faizulhaque-tp) [inline lib/repositories/collection.js]: no need to add catch.

**Pull request merged**